backend/src/game.ts

```diff
@@ -33,26 +36,86 @@ export interface Quiz {
 		note?: string;
 	};
 	questions: QuizQuestion[];
+	answers: Map<UserId, { time: number; answer: number }>;
 }
 
 export interface Game {
 	users: Map<UserId, User>;
 	hostId: UserId;
 	activeQuestion: number;
+	quizOpen: boolean;
 	quizData: Quiz;
 }
 
+// first key is gameId
+const games: Map<GameId, Game> = new Map();
+
 function getUsers(game: Game) {
 	return [...game.users.keys(), game.hostId];
 }
 
-function beginQuestion() {
+function endQuestion(gameId: GameId) {
+	const game = games.get(gameId);
+	if (!game) return;
+	const users = getUsers(game);
+	const userSockets = connections.get(gameId);
+	if (userSockets === undefined) {
+		// Player List Not in Connections
+		return;
+	}
+
+	users.forEach(function (value: UserId) {
+		let user = game.users.get(value);
+		if (!user) return;
+		let endResp = {
+			correct: game.quizData.questions[
+				game.activeQuestion
+			].correctAnswers.includes(user.answers[game.activeQuestion]),
+
+			correctAnswers:
+				game.quizData.questions[game.activeQuestion].correctAnswers,
+
+			score: user.scores.reduce((a, b) => a + b, 0),
+			scoreChange: user.scores[game.activeQuestion],
+			time: user.times[game.activeQuestion],
+		};
+		let resp = JSON.stringify(endResp);
+
+		let sock = userSockets.get(value);
+		if (sock === undefined) {
+			return;
+		}
+		if (sock.readyState === WebSocket.OPEN) {
+			sendMessage(sock, 'endQuestion', resp);
+		}
+	});
 	return;
 }
 
-// first key is gameId
-const games: Map<GameId, Game> = new Map();
-
+// Input: Game Object
+// beginQuestion sends each player and host the current active question
+function beginQuestion(gameId: GameId) {
+	const game = games.get(gameId);
+	if (!game) return;
+	const users = getUsers(game);
+	const userSockets = connections.get(gameId);
+	if (userSockets === undefined) {
+		// Player List Not in Connections
+		return;
+	}
+	// TODO: only send question data (dont send answers, etc)
+	const question = JSON.stringify(game.quizData.questions[game.activeQuestion]);
```

Comment:
I don't think you need to stringify here. `sendMessage` does that.

Reply:
Second this, small error from my original code.